internal/controller/linodemachine_controller.go

```diff
@@ -330,11 +330,13 @@ func (r *LinodeMachineReconciler) reconcilePreflightVPC(ctx context.Context, log
 	}
 	if err := machineScope.Client.Get(ctx, client.ObjectKeyFromObject(&linodeVPC), &linodeVPC); err != nil {
 		logger.Error(err, "Failed to fetch LinodeVPC")
-		if reconciler.RecordDecayingCondition(machineScope.LinodeMachine,
-			ConditionPreflightLinodeVPCReady, string(cerrs.CreateClusterError), err.Error(),
+		if reconciler.HasStaleCondition(machineScope.LinodeMachine,
+			ConditionPreflightLinodeVPCReady,
 			reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultClusterControllerReconcileTimeout)) {
+			conditions.MarkFalse(machineScope.LinodeMachine, ConditionPreflightLinodeVPCReady, string(cerrs.CreateClusterError), "", "%s", err.Error())
```

Comment:
Will this lead to `lastTransitionTime` getting updated for the condition? If yes, then in next iteration `HasStaleCondition()` will return false as it only looks at lastTransitionTime now. Previously, if it was already stale (by checking error severity), we were marking it as stale even if lastTransitionTime changed. Not sure here how it will behave.

Reply:
Ah good catch! Currently MarkFalse() will update the last transition time but I won't be using this func anymore in the follow up. I'll use Set() func which give more granular control over how to set conditions and I can just not set it then. 

But is this something to look out for in a lot of different places?

Reply:
Yup I confirmed that `lastTransitionTime` is not changed or update if the condition stays the same. MarkFalse() uses Set() func which has this comment over it:

```
// NOTE: If a condition already exists, the LastTransitionTime is updated only if a change is detected
// in any of the following fields: Status, Reason, Severity and Message.
```